modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,420 @@
     end
+    records
   end
 
-  def tldexpnd(targetdom,nssrv)
-    target = targetdom.scan(/(\S*)[.]\w*\z/).join
-    target.chomp!
-    if not nssrv.nil?
-      @res.nameserver=(nssrv)
-      @nsinuse = nssrv
+  def dns_wildcard(domain)
+    records = get_a("#{Rex::Text.rand_text_alpha(16)}.#{domain}")
+    if records.blank?
+      vprint_warning('dns wildcard is disable')
+      false
+    else
+      vprint_warning('dns wildcard is enable OR fake dns server')
+      true
     end
-    i, a = 0, []
-    tlds = [
-      "com", "org", "net", "edu", "mil", "gov", "uk", "af", "al", "dz",
-      "as", "ad", "ao", "ai", "aq", "ag", "ar", "am", "aw", "ac", "au",
-      "at", "az", "bs", "bh", "bd", "bb", "by", "be", "bz", "bj", "bm",
-      "bt", "bo", "ba", "bw", "bv", "br", "io", "bn", "bg", "bf", "bi",
-      "kh", "cm", "ca", "cv", "ky", "cf", "td", "cl", "cn", "cx", "cc",
-      "co", "km", "cd", "cg", "ck", "cr", "ci", "hr", "cu", "cy", "cz",
-      "dk", "dj", "dm", "do", "tp", "ec", "eg", "sv", "gq", "er", "ee",
-      "et", "fk", "fo", "fj", "fi", "fr", "gf", "pf", "tf", "ga", "gm",
-      "ge", "de", "gh", "gi", "gr", "gl", "gd", "gp", "gu", "gt", "gg",
-      "gn", "gw", "gy", "ht", "hm", "va", "hn", "hk", "hu", "is", "in",
-      "id", "ir", "iq", "ie", "im", "il", "it", "jm", "jp", "je", "jo",
-      "kz", "ke", "ki", "kp", "kr", "kw", "kg", "la", "lv", "lb", "ls",
-      "lr", "ly", "li", "lt", "lu", "mo", "mk", "mg", "mw", "my", "mv",
-      "ml", "mt", "mh", "mq", "mr", "mu", "yt", "mx", "fm", "md", "mc",
-      "mn", "ms", "ma", "mz", "mm", "na", "nr", "np", "nl", "an", "nc",
-      "nz", "ni", "ne", "ng", "nu", "nf", "mp", "no", "om", "pk", "pw",
-      "pa", "pg", "py", "pe", "ph", "pn", "pl", "pt", "pr", "qa", "re",
-      "ro", "ru", "rw", "kn", "lc", "vc", "ws", "sm", "st", "sa", "sn",
-      "sc", "sl", "sg", "sk", "si", "sb", "so", "za", "gz", "es", "lk",
-      "sh", "pm", "sd", "sr", "sj", "sz", "se", "ch", "sy", "tw", "tj",
-      "tz", "th", "tg", "tk", "to", "tt", "tn", "tr", "tm", "tc", "tv",
-      "ug", "ua", "ae", "gb", "us", "um", "uy", "uz", "vu", "ve", "vn",
-      "vg", "vi", "wf", "eh", "ye", "yu", "za", "zr", "zm", "zw", "int",
-      "gs", "info", "biz", "su", "name", "coop", "aero" ]
-    print_status("Performing Top Level Domain expansion using #{tlds.size} TLDs")
+  end
 
-    tlds.each do |tld|
-      query1 = @res.search("#{target}.#{tld}")
-      if (query1)
-        query1.answer.each do |rr|
-          print_status("Domain: #{target}.#{tld} Name: #{rr.name} IP address: #{rr.address} Record: A ") if rr.class == Net::DNS::RR::A
-          report_note(:host => @nsinuse.to_s,
-            :proto => 'udp',
-            :sname => 'dns',
-            :port => 53,
-            :type => 'dns.enum',
-            :update => :unique_data,
-            :data => "#{rr.address.to_s},#{target}.#{tld},A") if rr.class == Net::DNS::RR::A
-        end
-      end
-    end
+  def get_a(domain)
```

Comment:
Most of you `get_` methods are identical in the beginning in that they call `dns_query` and then return if the response is blank or there is not answer in the response.  IMO, you could/should refactor `dns_query` to do this work for you.  If you refactored `dns_query` to return a list of answers which will be an *empty list* if none are found for whatever reason, that would simplify each of you `get_*`